### PR TITLE
Fix text format

### PR DESCRIPTION
**Pull request by bkuehnert**
This PR documents the print-prim and print-enum functions, and makes the "indent" parameter more sensible.

Previously, indent was either a number or t. If it was a number, then indent that amount. If it was t, then don't indent. Now, indent is either a number or nil (if nil, don't indent).

### CODE REVIEW

**Comment by bkuehnert** [comment]
I'm not sure if this test failing is a bug or not. I think the test can fail as protobuf does not guarantee anything about the order of the fields, and the test asserts that the fields will be in a certain order. I'll try running this code on another computer and see if I can reproduce the failure.

**Comment by Slids** [comment]
Hey Ben, whats the state of this?

**Comment by bkuehnert** [comment]
> Hey Ben, whats the state of this?

I removed the test which could've been failing, and it's still failing on github. I'm not able to reproduce the error on the cloudtop, but will try tomorrow on another machine.

**Comment by Slids** [comment]
IT's feiling because it's sending skip field for #\-

**Comment by Slids** [comment]
Changing skip-field's ecase to 
(or #\: #\-)
I see

Original msg:
#S(CL-PROTOBUFS.TEST-PROTO:TEXT-FORMAT-TEST                                                                                        
   :%INT-FIELD 100                                                                                                                 
   :%SINT-FIELD -1                                                                                                                 
   :%UINT-FIELD 1                                                                                                                  
   :%FLOAT-FIELD 1.5                                                                                                               
   :%DOUBLE-FIELD 1.5d0                                                                                                            
   :%STRING-FIELD "A string"                                                                                                       
   :%STRING-FIELDS ("First" "Second")                                                                                              
   :%EMBEDDED-COMMENT-A 0                                                                                                          
   :%EMBEDDED-COMMENT-B 0                                                                                                          
   :%EMBEDDED-COMMENT-C 0                                                                                                          
   :%EMBEDDED-COMMENT-D 0                                                                                                          
   :%EMBEDDED-COMMENT-E 0                                                                                                          
   :%EMBEDDED-COMMENT-F 0                                                                                                          
   :%EMBEDDED-COMMENT-G 0                                                                                                          
   :%ONE-LEVEL-NESTING NIL                                                                                                         
   :%ENUM-VALS (:NONE :TWENTY-ONE)                                                                                                 
   :%BYTES NIL                                                                                                                     
   :%%IS-SET #*11111110000000010)
Parsed msg:
#S(CL-PROTOBUFS.TEST-PROTO:TEXT-FORMAT-TEST                                                                                        
   :%INT-FIELD 100                                                                                                                 
   :%SINT-FIELD -1                                                                                                                 
   :%UINT-FIELD 1                                                                                                                  
   :%FLOAT-FIELD 1.5                                                                                                               
   :%DOUBLE-FIELD 1.5d0                                                                                                            
   :%STRING-FIELD "A string"                                                                                                       
   :%STRING-FIELDS ("First" "Second")                                                                                              
   :%EMBEDDED-COMMENT-A 0                                                                                                          
   :%EMBEDDED-COMMENT-B 0                                                                                                          
   :%EMBEDDED-COMMENT-C 0                                                                                                          
   :%EMBEDDED-COMMENT-D 0                                                                                                          
   :%EMBEDDED-COMMENT-E 0                                                                                                          
   :%EMBEDDED-COMMENT-F 0                                                                                                          
   :%EMBEDDED-COMMENT-G 0                                                                                                          
   :%ONE-LEVEL-NESTING NIL                                                                                                         
   :%ENUM-VALS (:NONE NIL)                                                                                                         
   :%BYTES NIL                                                                                                                     
   :%%IS-SET #*11111110000000010)

**Comment by Slids** [comment]
Fix:
(defun print-enum (val enum field stream indent)                                                                                   
  "Print enum to stream                                                                                                            
                                                                                                                                   
Parameters:                                                                                                                        
  VAL: The enum value.                                                                                                             
  ENUM: The enum descriptor.                                                                                                       
  FIELD: The field which contains this value.                                                                                      
  STREAM: The stream to print to.                                                                                                  
  INDENT: Either a number or nil.                                                                                                  
          - If indent is a number, indent this print                                                                               
            by (+ indent 2) and write a newline at                                                                                 
            the end.                                                                                                               
          - If indent is nil, then do not indent and                                                                               
            do not write a newline."                                                                                               
  (when val                                                                                                                        
    (if indent                                                                                                                     
        (format stream "~&~VT~A: " (+ indent 2) (proto-name field))                                                                
        (format stream "~A: " (proto-name field)))                                                                                 
    (let* ((e (find (keywordify val)                                                                                               
                    (enum-descriptor-values enum)                                                                                  
                    :key #'enum-value-descriptor-value))                                                                           
           (value (and e (enum-value-descriptor-value e)))                                                                         
           (proto-keyword-value (substitute #\_ #\- (string value))))                                                              
                                                                                                                                   
      (format stream "~A" proto-keyword-value)                                                                                     
      (if indent                                                                                                                   
        (format stream "~%")                                                                                                       
        (format stream " ")))))

**Comment by Slids** [comment]
Warning:
CheckLint found errors.
Lines should be under 100 columns. (LINE-TOO-LONG)
	//depot/google3/third_party/lisp/cl_protobufs/text-format.lisp:114
Lines should be under 100 columns. (LINE-TOO-LONG)
	//depot/google3/third_party/lisp/cl_protobufs/text-format.lisp:121

File: text-format.lisp [hunk 1]
```diff
@@ -158,20 +180,6 @@ Parameters:
        (format stream "~A" (if val "true" "false")))
```
Comment (cgay) [inline text-format.lisp:180]: For later... I'm concerned that the way we output strings on line176 is wrong.  If the string contains embedded double quotes it won't work.  Maybe just using `~S` is enough, but I don't know if it's compatible with the textproto standard.

File: text-format.lisp [hunk 2]
```diff
@@ -48,7 +48,7 @@ Parameters:
                             (cond ((keywordp type)
                                    (doseq (v (read-slot object slot reader))
                                      (print-prim v type field stream
```
Comment (cgay) [inline text-format.lisp:50]: Another one not for this PR: `&key suppress-pretty-print` is terrible. We should change it to the positive sense: `&key (pretty-print t)`

Comment (bkuehnert) [reply text-format.lisp:50]: I think this is in scope of the PR. I'll change to that

File: tests/text-format-test.lisp [hunk 3]
```diff
@@ -61,6 +61,22 @@ Parameters:
...
+                                     :double-field 1.5d0
+                                     :string-field "A string"
+                                     :string-fields (list "First" "Second")
+                                     :enum-vals (list :none :twenty-one)))
+         (out-stream (make-string-output-stream)))
+    (print-text-format msg :stream out-stream)
+    (let* ((text (get-output-stream-string out-stream))
+           (msg-parse (proto:parse-text-format 'cl-protobufs.test-proto:text-format-test
+                                               :stream (make-string-input-stream text))))
+      (assert-true (proto:proto-equal msg-parse msg)))))
```
[the remark this answers is not in the document]
Comment (bkuehnert) [reply tests/text-format-test.lisp:79]: Oh.. Yes that appears to be what I was looking for.

File: text-format.lisp [hunk 4]
```diff
@@ -102,14 +103,34 @@ Parameters:
...
-                                                     (or suppress-pretty-print indent))))))
+                                                     (and (not suppress-pretty-print) indent))))))
+                                  ;; todo(benkuehnert): Sort output by key value/use specified format
+                                  ((typep msg 'map-descriptor)
+                                   (let ((key-class (map-descriptor-key-class msg))
+                                         (val-class (map-descriptor-val-class msg))
+                                         (val (read-slot object slot reader)))
+                                     (if suppress-pretty-print
+                                         (format stream "~A: {" (proto-name field))
+                                         (format stream "~&~VT~A: {~%" (+ 2 indent) (proto-name field)))
```
Comment (cgay) [inline text-format.lisp]: Apparently lines 114 and 121 are too long.  Needs fixing.

Comment (bkuehnert) [reply text-format.lisp]: This was fixed with 06a4e17e.

**Pull request merged**